app/services/builders/chartkick/base.rb

```diff
@@ -9,15 +9,43 @@ def initialize(entity_id, query)
       private
 
       def build_data(metrics)
-        metrics.each.inject({}) do |hash, metric|
+        metrics.inject({}) do |hash, metric|
           hash.merge!(
             metric.value_timestamp.strftime('%Y-%m-%d').to_s => value_in_hours_for(metric)
           )
         end
       end
 
-      def value_in_hours_for(metric)
-        (metric.value.to_f / 1.hour.seconds).round(2)
+      def build_distribution_data(entities)
+        entities.inject({}) do |hash, entity|
+          interval = resolve_interval(entity)
+          hash.merge!(
+            interval => value_in_hours_for(entity)
+          )
+        end
+      end
+
+      def value_in_hours_for(entity)
+        (entity.value.to_f / 1.hour.seconds).round(2)
+      end
+
+      def resolve_interval(entity)
+        entity_value = entity.value
+        if entity_value < 12
```

Comment:
what about using a `case` instead of if elsif?

Reply:
Tried it, but less performing.